moxin-frontend/src/chat/chat_panel.rs

```diff
@@ -296,16 +290,30 @@ impl Widget for ChatPanel {
         let store = scope.data.get_mut::<Store>().unwrap();
         let chat_history;
         let model_filename;
+        let initial_letter;
 
         if let Some(chat) = &store.current_chat {
             chat_history = chat.messages.clone();
             model_filename = chat.model_filename.clone();
+            initial_letter = model_filename
+                .chars()
+                .next()
+                .unwrap()
+                .to_uppercase()
+                .to_string();
         } else {
             chat_history = vec![];
             model_filename = "".to_string();
+            initial_letter = "".to_string();
         };
```

Comment:
we could get rid of the mutable variables with the [map_or](https://doc.rust-lang.org/std/option/enum.Option.html#method.map_or) method, though im totally find with the imperative approach, its in most places in this code and i find it completely readable, so take this just as a curiosity.

```rust
let (chat_history, model_filename, initial_letter) =
    store
	.current_chat
	.as_ref()
	.map_or((vec![], "".to_string(), "".to_string()), |chat| {
	    let model_filename = chat.model_filename.clone();
	    let initial_letter = model_filename
		.chars()
		.next()
		.unwrap_or_default()
		.to_uppercase()
		.to_string();
	    (chat.messages.clone(), model_filename, initial_letter)
	});
```